lib/_tls_wrap.js

```diff
@@ -247,7 +247,19 @@ function TLSSocket(socket, options) {
   // starting the flow of the data
   this.readable = true;
   this.writable = true;
-  this.read(0);
+
+  // Don't read until user-land had a chance to setup listeners
+  var self = this;
+  process.nextTick(function() {
+    // Socket already has some buffered data - emulate receiving it
+    if (socket && socket._readableState && socket._readableState.length) {
```

Comment:
I know this has just been moved, but we really need to standardize an API for querying the size of a stream's buffer.